File: Python/Reverse Bits.py
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        ret = 0
        pt = 31
        while n:
            ret |= n&1 and 1<<pt or 0
            n >>= 1
            pt -= 1
        return ret

File: java/Reverse_Bits.java
//Bit manipulation
// Time O(1), space O(1).

public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int result = 0;
        for (int i = 0; i < 32; ++i)
            result = (result<<1)|((n>>i)&0x1);
        return result;
    }
}
